atomix/cluster/src/main/java/io/atomix/raft/snapshot/impl/FileBasedReceivedSnapshot.java

```diff
@@ -192,27 +192,29 @@ public PersistedSnapshot persist() {
     final var files = directory.toFile().listFiles();
     Objects.requireNonNull(files, "No chunks have been applied yet");
 
-    if (files.length != expectedTotalCount) {
+    if (expectedTotalCount != Integer.MIN_VALUE && files.length != expectedTotalCount) {
       throw new IllegalStateException(
           String.format(
               "Expected '%d' chunk files for this snapshot, but found '%d'. Files are: %s.",
               expectedSnapshotChecksum, files.length, Arrays.toString(files)));
     }
 
-    final var filePaths =
-        Arrays.stream(files).sorted().map(File::toPath).collect(Collectors.toList());
-    final long actualSnapshotChecksum;
-    try {
-      actualSnapshotChecksum = ChecksumUtil.createCombinedChecksum(filePaths);
-    } catch (final IOException e) {
-      throw new UncheckedIOException("Unexpected exception on calculating snapshot checksum.", e);
-    }
-
-    if (actualSnapshotChecksum != expectedSnapshotChecksum) {
-      throw new IllegalStateException(
-          String.format(
-              "Expected snapshot checksum %d, but calculated %d.",
-              expectedSnapshotChecksum, actualSnapshotChecksum));
+    if (expectedSnapshotChecksum != Long.MIN_VALUE) {
```

Comment:
The same here - currently I would ask why it should be min value. But if you add a constant which describes this a bit better I think it would be clearer

Reply:
iirc I think SBE uses this as the null value, so I assume this came from there - naming the constants is probably anyway the better solution :+1: